pkg/util/fs.go

```diff
@@ -193,7 +273,31 @@ func (h *fs) Open(filename string) (io.ReadCloser, error) {
 	return os.Open(filename)
 }
 
-// Write opens a file and writes data to it, returning error if such occurred
+// WriteFile opens a file and writes data to it, returning error if such
+// occurred
 func (h *fs) WriteFile(filename string, data []byte) error {
 	return ioutil.WriteFile(filename, data, 0700)
 }
+
+// Walk walks the file tree rooted at root, calling walkFn for each file or
+// directory in the tree, including root.
+func (h *fs) Walk(root string, walkFn filepath.WalkFunc) error {
+	wrapper := func(path string, info os.FileInfo, err error) error {
+		if runtime.GOOS == "windows" && err == nil {
+			h.m.Lock()
+			info = h.enrichFileInfo(path, info)
+			h.m.Unlock()
+		}
+		return walkFn(path, info, err)
+	}
+	return filepath.Walk(root, wrapper)
+}
+
+func (h *fs) enrichFileInfo(path string, fi os.FileInfo) os.FileInfo {
+	// call me holding the mutex
```

Comment:
I see `h.enrichFileInfo(path, info)` is called from multiples places, we should definitely lock/unlock here and not in every call site.